content/en/docs/Reference/file-based-catalogs.md

```diff
@@ -0,0 +1,586 @@
+  docker push registry.org/my-catalog:latest-abc123-def456
+  
+  # Create a new catalog from the heads of an existing catalog, then build and push the image like above.
+  opm alpha diff registry.org/my-catalog:def456 -o yaml > my-catalog-index/index.yaml
+  docker build -t registry.org/my-catalog:headsonly-def456 -f index.Dockerfile .
+  docker push registry.org/my-catalog:headsonly-def456
+
+Flags:
+      --ca-file string   the root Certificates to use with this command
+      --debug            enable debug logging
+  -h, --help             help for diff
+  -o, --output string    Output format (json|yaml) (default "yaml")
+
+Global Flags:
+      --skip-tls   skip TLS certificate verification for container image registries while pulling bundles or index
+```
+
+### `opm alpha generate dockerfile`
+
+```
+Generate a Dockerfile for a declarative config index.
+
+This command creates a Dockerfile in the same directory as the <dcRootDir>
+(named <dcDirName>.Dockerfile) that can be used to build the index. If a
+Dockerfile with the same name already exists, this command will fail.
+
+When specifying extra labels, note that if duplicate keys exist, only the last
+value of each duplicate key will be added to the generated Dockerfile.
+
+Usage:
+  opm alpha generate dockerfile <dcRootDir> [flags]
+
+Flags:
+  -i, --binary-image string    Image in which to build catalog. (default "quay.io/operator-framework/upstream-opm-builder")
+  -l, --extra-labels strings   Extra labels to include in the generated Dockerfile. Labels should be of the form 'key=value'.
+  -h, --help                   help for dockerfile
+
+Global Flags:
+      --skip-tls   skip TLS certificate verification for container image registries while pulling bundles or index
+```
+
+## Guidelines
+
+### Immutable bundles
+
+OLM's general advice is that bundle images and their metadata should be treated as immutable. If a broken bundle has 
+been pushed to an index, you must assume that at least one of your users has upgraded to that bundle. Based on that
+assumption, you must release another bundle with an upgrade edge from the broken bundle to ensure users with the broken
+bundle installed receive an upgrade. OLM will not reinstall an installed bundle if the contents of that bundle are
+updated in the index.
+
+However, there are some cases where a change in the index metadata is preferred. For example:
+- Channel promotion - if you already released a bundle and later decide that you'd like to add it to another channel,
+  simply add an `olm.channel` property to the `olm.bundle`
+- New upgrade edges - if you release a new 1.2.z (e.g. 1.2.4), but 1.3.0 is already released, you can update the index
+  metadata for 1.3.0 to skip 1.2.4.
+  
+### Use of source control
+
+OLM highly recommends storing index metadata in source control and treating the source-controlled metadata as the source
+of truth. Updates to index images should:
```

Comment:
It may be nice to link out to a repository that has this directory structure in the future.

Reply:
Added a TODO comment.